### PR TITLE
Link imported ObjC forward declarations to their Swift definition

### PR DESCRIPTION
**Pull request by NuriAmari**
On occasion, an Objective-C header will forward declare an interface or protocol that is defined elsewhere in Swift. If the definition were in a Clang module instead, the forward declaration would be linked to its definition. Prior to this patch, this was not the case for @objc exposed Swift declarations. This patch fixes that.

### CODE REVIEW

**Comment by NuriAmari** [comment]
@swift-ci Please test

**Comment by NuriAmari** [comment]
@swift-ci Please smoke test macOS platform

**Comment by NuriAmari** [comment]
@swift-ci Please smoke test macOS platform

**Comment by NuriAmari** [comment]
@swift-ci Please test macOS platform

**Comment by NuriAmari** [comment]
@swift-ci Please test

File: include/swift/Basic/LangOptions.h [hunk 1]
```diff
@@ -808,6 +808,10 @@ namespace swift {
     /// instead of dropped altogether when possible.
     bool ImportForwardDeclarations = false;
 
+    /// If true, the compiler will link forward declarations in Objective-C
+    /// that represent an @objc exposed Swift definition to their definition.
+    bool ResovleObjCForwardDeclarationsOfSwiftTypes = false;
```
Comment (drodriguez) [inline include/swift/Basic/LangOptions.h:813]: Nit: typo. Change to `ResolveObjCForwardDeclarationsOfSwiftTypes`

**Comment by NuriAmari** [comment]
@swift-ci Please test

**Comment by NuriAmari** [comment]
@swift-ci Please test source compatibility

**Comment by NuriAmari** [comment]
@swift-ci Please test

**Pull request closed without merging**